addons/point_of_sale/models/stock_picking.py

```diff
@@ -138,6 +138,13 @@ def _compute_hide_reservation_method(self):
             if picking_type == picking_type.warehouse_id.pos_type_id:
                 picking_type.hide_reservation_method = True
 
+    def action_archive(self):
+        for picking_type in self:
+            pos_configs = self.env['pos.config'].search([('picking_type_id', '=', picking_type.id)])
+            if pos_configs:
+                raise UserError(_("You cannot archive this Operation Type as it is used by a POS configuration."))
+        return super(StockPickingType, self).action_archive()
```

Comment:
It would be better to create an ORM constraint, because here I could directly write on `active` field (i.e. archiving without use `action_archive` method) and still get the issue